cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -598,7 +605,8 @@ func (p *BicepProvider) Deploy(ctx context.Context) (*provisioning.DeployResult,
 	}
 
 	deploymentTags := map[string]*string{
-		azure.TagKeyAzdEnvName: to.Ptr(p.env.Name()),
+		azure.TagKeyAzdEnvName:   to.Ptr(p.env.Name()),
+		azure.TagKeyAzdStageName: &p.stage,
```

Comment:
What value are we setting the `azd-stage-name` tag to for the "default" stage (the one provisioned by running `azd provision`)?

Reply:
It would be set as "default", whatever the default value is.

On matching, the new deployment would match on tags. However, for older deployments already out there, we'd match based on env name alone.